docs/docs/how_to/how-to-recursion.md

```diff
@@ -0,0 +1,178 @@
+---
+title: Recursion with NoirJS
+description: Learn how to implement recursion with NoirJS, a powerful tool for creating smart contracts on the EVM blockchain. This guide assumes familiarity with NoirJS, solidity verifiers, and the Barretenberg proving backend. Discover how to generate both final and intermediate proofs using `noir_js` and `backend_barretenberg`.
+keywords:
+  [
+    "NoirJS",
+    "EVM blockchain",
+    "smart contracts",
+    "recursion",
+    "solidity verifiers",
+    "Barretenberg backend",
+    "noir_js",
+    "backend_barretenberg",
+    "intermediate proofs",
+    "final proofs",
+    "nargo compile",
+    "json import",
+    "recursive circuit",
+    "recursive app"
+  ]
+sidebar_position: 1
+---
+
+We're assuming you already read the [recursion explainer](../explainers/explainer-recursion.md), and that you already built a recursive circuit following [the reference](../noir/standard_library/recursion.md).
```

Comment:
When I try to copy paste the reference, this is what I see 
![image](https://github.com/noir-lang/noir/assets/18372439/efbc5668-e1b9-423f-9784-69bdc8baaa3c)

am I doing something wrong?

Reply:
I think you need to do `.as_slice()` as that is what the builtin expects. The example should be updated though so others don't hit the same.

Reply:
damn that's right, I'll fix this in this PR